### PR TITLE
samples(adt): Update ctor snippets, and update sample options

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/ComponentSamples.cs [hunk 1]
```diff
@@ -51,7 +51,7 @@ public async Task RunSamplesAsync()
             await DigitalTwinsClient.CreateDigitalTwinAsync(twinId, twinPayload).ConfigureAwait(false);
             Console.WriteLine($"Created digital twin {twinId}.");
 
-            #region Snippet:DigitalTwinSampleUpdateComponent
+            #region Snippet:DigitalTwinsSampleUpdateComponent
```
Comment (drwill-ms) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/ComponentSamples.cs:54]: Renaming all snippet prefixes to DigitalTwin_s_

File: sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/Options.cs [hunk 2]
```diff
@@ -2,57 +2,44 @@
...
+        public string AdtEndpoint { get; set; }
 
-        [Option("adtEndpoint", Required = true, HelpText = "Digital twins service endpoint")]
-        public string AdtEndpoint { get; }
+        [Option('i', "clientId", Required = true, HelpText = "Client Id of the application Id to login, or the application Id used to log the user in.")]
+        public string ClientId { get; set; }
 
-        [Option("clientId", Required = true, HelpText = "Application client Id")]
-        public string ClientId { get; }
+        [Option('m', "loginMethod", Required = false, Default = "AppId", HelpText = "Choose between: AppId, User.")]
```
Comment (drwill-ms) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/Options.cs:23]: Add support for interactive login

File: sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/Program.cs [hunk 3]
```diff
@@ -20,17 +21,39 @@ public static async Task Main(string[] args)
...
+                .WithParsed(parsedOptions =>
+                    {
+                        options = parsedOptions;
+                    })
+                .WithNotParsed(errors =>
+                    {
+                        Environment.Exit(1);
+                    });
+
+            if (options.GetLoginMethod() == LoginMethod.AppId
```
Comment (drwill-ms) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/Program.cs:33]: Special handling for optionally required parameter.

Comment (azabbasi) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/Program.cs:33]: you can use mutually exclusive sets for your options so this case automatically gets handled for you.

**Review by azabbasi — APPROVED** [review]

**Review by vinagesh — APPROVED** [review]

File: sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/Program.cs (hunk 3 above)
Comment (drwill-ms) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/Program.cs:33]: Got an example? I tried but couldn't find one.

**Review by barustum — APPROVED** [review]

**Pull request merged**